### PR TITLE
Update go 18

### LINKED ISSUE #154: update ci to go 1.18
**Describe the solution you'd like**
update to go 1.18

### PR DESCRIPTION
**Pull request by bdlb77**
**What this PR does / why we need it**:

This PR updates Golang to v1.18 throughout the Project. 

**Which issue(s) this PR fixes**:

Fixes #154 

**Special notes for your reviewer**:

Update Golang in:

- Dockerfiles
- go.mod 
- CI Pipeline

> Note: `go.mod` indirect requires are in own block form v1.17 and on.. `go.mod` was version v1.16 before this PR, so that is why there are many changes when running `go get -u && go mod tidy`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/Azure/eraser
 
-go 1.16
+go 1.18
```
Comment (sozercan) [inline go.mod]: we don't need to update this as we don't rely on go 1.18 features yet

**Comment by sozercan** [comment]
i think we have a few Dockerfiles that needs to be updated, let's use `golang:1.18-bullseye` image for those

File: go.mod (hunk 1 above)
Comment (bdlb77) [reply go.mod]: Reverted this back to `1.16`

**Comment by bdlb77** [comment]
> i think we have a few Dockerfiles that needs to be updated, let's use `golang:1.18-bullseye` image for those

Updated the images to `golang:1.18-bullseye` ... In both

`/Dockerfile` and `build/tooling/Dockerfile`

**Review by sozercan — APPROVED** [review]
Thank you!! LGTM

**Comment by bdlb77** [comment]
Thank you @sozercan  for reviewing and feedback! I'm not able to Merge so I will wait for your team to do it :)

**Pull request merged**